### PR TITLE
add user stuff - #1

### PR DESCRIPTION
**Pull request by meta1203**

### CODE REVIEW

File: src/main/java/com/collegecompendium/backend/models/User.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	@GeneratedValue(strategy = GenerationType.UUID)
+    private final String id;
+	@NotNull
+	@Email
+	@Column(length = 1024)
+    private String email;
+	@NotNull
+	@NotEmpty
+	@Column(length = 128)
+	private String username;
```
Comment (Korbrent) [inline src/main/java/com/collegecompendium/backend/models/User.java:35]: Might want to add a constraint later ensuring usernames are unique, this should be doable using `@Column(nullable = false, unique = true)` annotation. 

I'll add this on my next pull request.

File: src/main/java/com/collegecompendium/backend/repositories/UserRepository.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.collegecompendium.backend.repositories;
+import org.springframework.data.repository.CrudRepository;
+
+import com.collegecompendium.backend.models.User;
+
+public interface UserRepository extends CrudRepository<User, String>{
```
Comment (Korbrent) [inline src/main/java/com/collegecompendium/backend/repositories/UserRepository.java:6]: Do we need the `@Repository` annotation here?

File: src/main/java/com/collegecompendium/backend/models/User.java (hunk 1 above)
Comment (Korbrent) [reply src/main/java/com/collegecompendium/backend/models/User.java:35]: Done

File: src/main/java/com/collegecompendium/backend/repositories/UserRepository.java (hunk 2 above)
Comment (Korbrent) [reply src/main/java/com/collegecompendium/backend/repositories/UserRepository.java:6]: Done

**Review by Korbrent — COMMENTED** [review]
Completed my review, added minor changes. Awaiting secondary review

**Review by ENoll78 — APPROVED** [review]

**Pull request merged**